### PR TITLE
Add Collector executor support

### PR DESCRIPTION
**Pull request by vmarmol**
/cc @rjnagal

### CODE REVIEW

File: manager/container.go [hunk 1]
```diff
@@ -226,12 +232,24 @@ func (c *containerData) housekeeping() {
 			}
 		}
 
-		// Schedule the next housekeeping. Sleep until that time.
+		// Run custom collectors.
+		nextCollectionTime, err := c.collectorManager.Collect()
```
Comment (rjnagal) [inline manager/container.go]: check for nil. For most cases, we want to have collectorManager as nil.

Comment (vmarmol) [inline manager/container.go]: I was more thinking that it'd never be nil although there may be no collectors. I want to re-implement stats as a collector so there should be at least one collector.

Comment (vmarmol) [inline manager/container.go]: WDYT?

Comment (rjnagal) [inline manager/container.go]: ok, SG.

**Comment by rjnagal** [comment]
LGTM

**Pull request merged**